includes/multi-currency/Compatibility/BaseCompatibility.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Class BaseCompatibility
+ *
+ * @package WCPay\MultiCurrency\Compatibility
+ */
+
+namespace WCPay\MultiCurrency\Compatibility;
+
+use WCPay\MultiCurrency\MultiCurrency;
+use WCPay\MultiCurrency\Utils;
+
+/**
+ * Class that sets up base options for compatibility classes.
+ */
+class BaseCompatibility {
+
+	const FILTER_PREFIX = 'wcpay_multi_currency_';
+
+	/**
+	 * MultiCurrency class.
+	 *
+	 * @var MultiCurrency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Utils class.
+	 *
+	 * @var Utils
+	 */
+	protected $utils;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param MultiCurrency $multi_currency MultiCurrency class.
+	 * @param Utils         $utils          Utils class.
+	 */
+	public function __construct( MultiCurrency $multi_currency, Utils $utils ) {
+		$this->multi_currency = $multi_currency;
+		$this->utils          = $utils;
+		$this->init();
+	}
+
+	/**
+	 * Init the class.
+	 *
+	 * @return void
+	 */
+	protected function init() {
```

Comment:
This can also be marked as abstract.

Reply:
Updated in https://github.com/Automattic/woocommerce-payments/pull/3259/commits/0f0dff55744bcb0d3c04dea2d1a6b7da5e6147e7